### PR TITLE
add AggFunctionManager

### PR DESCRIPTION
**Pull request by czpmango**
Remove the dependence of graph on `AggregateFunction`.

It will bring the following two benefits:
 - Simplify the development complexity of adding aggregate functions. Only need to add the implementation of specific functions in `AggFunctionManager`, just like what `FunctionManager` did.
 - Serving the implementation of `dynamic loading` later.

Depended by [https://github.com/vesoft-inc/nebula-graph/pull/802](https://github.com/vesoft-inc/nebula-graph/pull/802)

### CODE REVIEW

**Review by jievince — DISMISSED** [review]

File: src/common/function/AggFunctionManager.cpp [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+Status AggFunctionManager::find(const std::string &func) {
+    auto result = instance().getInternal(func);
+    NG_RETURN_IF_ERROR(result);
+    return Status::OK();
+}
+
+StatusOr<AggFunctionManager::AggFunction> AggFunctionManager::getInternal(std::string func) const {
+    std::transform(func.begin(), func.end(), func.begin(), ::toupper);
+    folly::RWSpinLock::ReadHolder holder(lock_);
```
Comment (CPWstatic) [inline src/common/function/AggFunctionManager.cpp:318]: This `lock` is not necessary, we had the same problem in FunctionManager, please fix it.

Comment (czpmango) [reply src/common/function/AggFunctionManager.cpp:318]: ok.

**Review by jievince — APPROVED** [review]

**Review by CPWstatic — DISMISSED** [review]
LGTM.

File: src/common/function/test/AggFunctionManagerTest.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    google::SetStderrLogging(google::INFO);
+
+    auto result = nebula::time::TimeUtils::initializeGlobalTimezone();
+    if (!result.ok()) {
+        LOG(FATAL) << result;
+    }
+
+    DLOG(INFO) << "Timezone: " << nebula::time::TimeUtils::getGlobalTimezone().stdZoneName();
+    DLOG(INFO) << "Timezone offset: " <<
+        nebula::time::TimeUtils::getGlobalTimezone().utcOffsetSecs();
```
Comment (laura-ding) [inline src/common/function/test/AggFunctionManagerTest.cpp:109]: Your test does not use the TimeUtil, no need to init timezone.

File: src/common/function/test/CMakeLists.txt [hunk 3]
```diff
@@ -20,3 +20,20 @@ nebula_add_test(
...
+        agg_function_manager_test
+    SOURCES
+        AggFunctionManagerTest.cpp
+    OBJECTS
+        $<TARGET_OBJECTS:agg_function_manager_obj>
+        $<TARGET_OBJECTS:base_obj>
+        $<TARGET_OBJECTS:datatypes_obj>
+        $<TARGET_OBJECTS:time_utils_obj>
+        $<TARGET_OBJECTS:time_obj>
+        $<TARGET_OBJECTS:fs_obj>
```
Comment (laura-ding) [inline src/common/function/test/CMakeLists.txt:34]: Object is not required

File: src/common/function/test/AggFunctionManagerTest.cpp (hunk 2 above)
Comment (laura-ding) [inline src/common/function/test/AggFunctionManagerTest.cpp:109]: Your test is not used TimeUtil, so here is no need to init timezone

Comment (czpmango) [reply src/common/function/test/AggFunctionManagerTest.cpp]: Already deleted.

File: src/common/function/test/CMakeLists.txt (hunk 3 above)
Comment (czpmango) [reply src/common/function/test/CMakeLists.txt]: ok

File: src/common/function/AggFunctionManager.cpp [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+AggFunctionManager::AggFunctionManager() {
+    // TODO: private ctor to construct built-in aggregate functions
+    {
+        auto &func = functions_[""];
+        func = [](AggData* aggData, const Value& val) {
+            aggData->setResult(val);
+        };
+    }
+    {
+        auto &func = functions_["COUNT"];
```
Comment (laura-ding) [inline src/common/function/AggFunctionManager.cpp:30]: the function name can unify to lowercase with other function names

**Review by laura-ding — APPROVED** [review]

File: src/common/function/AggFunctionManager.cpp (hunk 4 above)
Comment (czpmango) [reply src/common/function/AggFunctionManager.cpp:30]: This can be used to distinguish between ordinary functions and aggregate functions.

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**